src/fast_scanning.jl

```diff
@@ -0,0 +1,136 @@
+sharpness{CT<:Images.NumberLike,N}(img::AbstractArray{CT,N}) = var(imfilter(img, Kernel.Laplacian(ntuple(i->true, Val{N}))))
+
+function adaptive_thres{CT<:Images.NumberLike,N}(img::AbstractArray{CT,N}, block::NTuple{N,Int})
+    threshold = zeros(Float64, block)
+    block_length = CartesianIndex(ntuple(i->ceil(Int,length(indices(img,i))/block[i]),Val{N}))
+    net_s = sharpness(img)
+    net_var = var(img)
+    net_end = last(CartesianRange(indices(img)))
+    for i in CartesianRange(block)
+        si = CartesianIndex(ntuple(j->(i[j]-1)*block_length[j]+1,Val{N}))
+        ei = min(si + block_length - 1, net_end)
+        wi = view(img, map((i,j)->i:j, si.I, ei.I)...)
+        threshold[i] = 0.02 + min(sharpness(wi)/net_s*0.04,0.1) + min(var(wi)/net_var*0.04,0.1)
+    end
+    threshold
+end
+
+getscalar{T<:Real,N}(A::AbstractArray{T,N}, i::CartesianIndex{N}, block_length::CartesianIndex{N}) =
+    A[CartesianIndex(ntuple(j->(i[j]-1)÷block_length[j]+1, Val{N}))]
+
+getscalar(a::Real, i...) = a
+
+"""
+    seg_img = fast_scanning(img, threshold, [diff_fn])
+
+Segments the N-D image using a fast scanning algorithm and returns a
```

Comment:
Come to think of it, a citation for most of our algorithms might be a good idea. I've not done that systematically in Images.jl, unfortunately.